Source/Utilities/DOMReady.js

```diff
@@ -77,6 +77,7 @@ if ('onreadystatechange' in document) document.addListener('readystatechange', c
 else shouldPoll = true;
 
 if (shouldPoll) poll();
+if (!ready) check();
```

Comment:
for the old php file, bear in mind the original case scenario which I described here:

http://webreflection.blogspot.co.uk/2006/09/too-much-simple-domcontentloaded.html?showComment=1158586140000#c115858618235493470

It would be possible to simulate the php flush using a simple write in nodejs like in this stackoverflow question:
http://stackoverflow.com/questions/16184103/how-to-flush-chunks-of-arbitrary-sizes-in-nodejs

also I think the fix should be done here https://github.com/SergioCrisostomo/mootools-core/blob/fix-2682/Source/Utilities/DOMReady.js#L76 basically if the event is supported we have to check if the document readystate is completed, if it is, we can just run the domready